test/unit/extra_fields_test.rb

```diff
@@ -1,279 +1,230 @@
 require 'test_helper'
 
-
 class ExtraFieldsTest < ActiveSupport::TestCase
 
-  setup do
-    @provider = FactoryBot.create :provider_account
-    @buyer = FactoryBot.create :buyer_account, :provider_account => @provider
-
-    FactoryBot.create(:fields_definition, :account => @provider, :target => "Account",
-            :name => "provider_extra_field", :required => true)
-    FactoryBot.create(:fields_definition, :account => Account.master,
-            :target => "Account",
-            :name => "master_extra_field", :required => true)
+  def setup
+    @provider = FactoryBot.create(:provider_account, provider_account: master_account)
+    @buyer = FactoryBot.create(:buyer_account, provider_account: @provider)
 
     @provider.reload
     @buyer.reload
   end
 
-  test 'extra_fields not validated by default' do
-    assert @buyer.valid?
-  end
+  class FieldsDefinitionTest < ExtraFieldsTest
 
-  test '#validate_fields!' do
-    # extra fields are validated on demand only
-    @buyer.validate_fields!
-    User.current = @buyer.admins.first
+    def setup
+      super
 
-    refute @buyer.valid?
-    assert_not_empty @buyer.errors["provider_extra_field"]
-  end
+      @provider_field = Account.optional_fields.first
```

Comment:
That are some assumptions for first and last

Reply:
@hallelujah Like before this PR, right? What do you propose instead?

Reply:
Making it more explicit that optional_fields have multiple entries
Either by a hash or new methods?

Reply:
We spoke about this and decided that I will just make a Jira because it is too much work that does not worth it for now considering that the Rails 5 PR depends on this one :+1: